core/src/main/scala/sttp/tapir/generic/Configuration.scala

```diff
@@ -1,15 +1,17 @@
-package sttp.tapir.generic
+ package sttp.tapir.generic
 
 import java.util.regex.Pattern
+import magnolia.TypeName
 
-final case class Configuration(toEncodedName: String => String, discriminator: Option[String]) {
+final case class Configuration(toEncodedName: String => String, discriminator: Option[String], toEncodedSubtypeName: TypeName => String) {
```

Comment:
I think here we'd need to avoid depending on magnolia's class, as this is kind of an implementation detail. Maybe it would make sense to use `SName => String` here?